src/cbmc_starter_kit/template-for-repository/proofs/Makefile.common

```diff
@@ -252,12 +252,12 @@ CBMC_FLAG_SIGNED_OVERFLOW_CHECK ?= --signed-overflow-check
 CBMC_FLAG_UNDEFINED_SHIFT_CHECK ?= --undefined-shift-check
 CBMC_FLAG_UNSIGNED_OVERFLOW_CHECK ?= --unsigned-overflow-check
 CBMC_FLAG_UNWINDING_ASSERTIONS ?= --unwinding-assertions
-CBMC_FLAG_UNWIND ?= --unwind 1
+CBMC_DEFAULT_UNWIND ?= --unwind 1
```

Comment:
🍓

Reply:
Same answer as 🍓. Plus, this is an internal variable that it's only used in the `Makefile.common`.

Reply:
It actually was never used in Makefile.common before. So we are now passing `--unwind 1` where we were not before.

I'm very confused.

How did the proofs ever work when --unwind 1 was not being passed to them?

It seems that there is now a risk that we break proofs by newly passing --unwind 1 because the proofs did not have that restriction before, and so adding this unwind bound may cause an unwinding assertion to fail?

Reply:
@karkhaz It was always used before. This recent [commit](https://github.com/model-checking/cbmc-starter-kit/commit/36acd4385cd01a3ee449b334bdf9ac472146a0b1) that removed from our `Makefile.common`.

Reply:
To be more specific, this [commit](https://github.com/model-checking/cbmc-starter-kit/pull/150/commits/691274ad64b2b4146f1e3116ea0cf832ae7b046b).

Reply:
In earlier versions of the starter kit (up to the one before this PR)

The `--unwind 1` flag was passed to `cbmc` trough in `CBMC_FLAGS`

```
CBMC_FLAG_UNWIND ?= --unwind 1
...
CBMCFLAGS += $(CBMC_FLAG_UNWIND) $(CBMC_UNWINDSET) $(CBMC_FLAG_FLUSH)
```

And the unwinding assertions flag was passed together with CHECK_FLAGS
```
CBMC_FLAG_UNWINDING_ASSERTIONS ?= --unwinding-assertions
...
CHECKFLAGS += $(CBMC_FLAG_UNWINDING_ASSERTIONS)
```

So all proofs were running using `--unwind 1` as default unless they were already processed using `UNWINDSET`.   

In this PR this is still how things happen when no contracts are used. 

When old contracts are used, user loops are unwound before contract instrumentation, and CPROVER library loops are unwound by cbmc at the end, and -- unwind 1 is also given to cbmc.

When DFCC contracts are used, the user loops and CPROVER library loops are unwound before contract instrumentation and --unwind 1 is applied as well before contract instrumentation, which results in all loops being bounded just like when no contracts are used or when old contracts are used, except for the loops introduced by the dynamic frames instrumentation which are dynamically unwound during symex (and always terminate).